### PR TITLE
Project Recovery: add execCount to the comment for call to GeneratePythonScript

### LINKED ISSUE #25014: Unexpected exception in project recovery
### Expected behavior
Project recovery should work

### Actual behavior
Project recovery throws an unexpected exception when attempting to recover

project_recovery_test1_crash

### Steps to reproduce the behavior
Follow the instructions in test1 in the [unscripted testing](http://developer.mantidproject.org/Testing/ErrorReporter-ProjectRecovery/ProjectRecoveryTesting.html). When trying to recover after crashing after the second, longer script, you get the exception

### Platforms affected
Windows 10 (bug doesn't occur on Ubuntu)

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
Project recovery would in some cases be unable to recover because algorithms ran so quickly that the Operating system's time could not be rendered accurately enough to distinguish this fact. As added ordered attempts I have added the execCount to the comment section of script builds.

**To test:**
Perform the project recovery tests on a release build on a newly restarted PC.
http://developer.mantidproject.org/Testing/ErrorReporter-ProjectRecovery/ProjectRecoveryTesting.html

Fixes #25014 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by gemmaguest — DISMISSED** [review]
Code changes look good but I think there is a problem sorting by string rather than numeric values e.g.  10 gets ordered before 3 as shown below. An easy fix for this might be to pad it with zeros? Although the difficulty there might be knowing how many are needed?

```
CreateWorkspace(OutputWorkspace='0Rebinned', DataX='0,1,2,3,4,5,6,7,8,9,10,11', DataY='0,1,2,3,4,5,6,7,8,9,10,11', DataE='0,1,2,3,4,5,6,7,8,9,10,11', NSpec=4) # 2019-03-20T17:21:54.105411000 execCount: 1
RenameWorkspace(InputWorkspace='0Rebinned', OutputWorkspace='1Rebinned') # 2019-03-20T17:21:54.120583000 execCount: 2
RenameWorkspace(InputWorkspace='8Rebinned', OutputWorkspace='9Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 10
RenameWorkspace(InputWorkspace='9Rebinned', OutputWorkspace='10Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 11
RenameWorkspace(InputWorkspace='10Rebinned', OutputWorkspace='11Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 12
RenameWorkspace(InputWorkspace='11Rebinned', OutputWorkspace='12Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 13
RenameWorkspace(InputWorkspace='12Rebinned', OutputWorkspace='13Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 14
RenameWorkspace(InputWorkspace='13Rebinned', OutputWorkspace='14Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 15
RenameWorkspace(InputWorkspace='14Rebinned', OutputWorkspace='15Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 16
RenameWorkspace(InputWorkspace='1Rebinned', OutputWorkspace='2Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 3
RenameWorkspace(InputWorkspace='2Rebinned', OutputWorkspace='3Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 4
RenameWorkspace(InputWorkspace='3Rebinned', OutputWorkspace='4Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 5
RenameWorkspace(InputWorkspace='4Rebinned', OutputWorkspace='5Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 6
RenameWorkspace(InputWorkspace='5Rebinned', OutputWorkspace='6Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 7
RenameWorkspace(InputWorkspace='6Rebinned', OutputWorkspace='7Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 8
RenameWorkspace(InputWorkspace='7Rebinned', OutputWorkspace='8Rebinned') # 2019-03-20T17:21:54.122090000 execCount: 9
RenameWorkspace(InputWorkspace='15Rebinned', OutputWorkspace='16Rebinned') # 2019-03-20T17:21:54.137702000 execCount: 17
RenameWorkspace(InputWorkspace='16Rebinned', OutputWorkspace='17Rebinned') # 2019-03-20T17:21:54.138706000 execCount: 18
RenameWorkspace(InputWorkspace='17Rebinned', OutputWorkspace='18Rebinned') # 2019-03-20T17:21:54.138706000 execCount: 19
RenameWorkspace(InputWorkspace='18Rebinned', OutputWorkspace='19Rebinned') # 2019-03-20T17:21:54.138706000 execCount: 20
RenameWorkspace(InputWorkspace='19Rebinned', OutputWorkspace='20Rebinned') # 2019-03-20T17:21:54.138706000 execCount: 21
```

**Comment by gemmaguest** [comment]
I've unassigned myself because I'm on holiday now for a few days

**Review by martyngigg — DISMISSED** [review]
This looks fine. The rename workspace calls now come out in the order they are expected to come out in. :+1:

File: docs/source/release/v4.0.0/mantidplot.rst [hunk 1]
```diff
@@ -49,5 +49,6 @@ Bugfixes
 - A bug has been fixed in Indirect I(Q,t) interface when analyzing ``IN16B`` reduced data.
 - Stop MantidPlot from attempting to contour plot/colorfill plot when a workspace has a TextAxis.
 - The general fitting interfaces do not display fit functions that are not supposed to be used with those interfaces.
+- Project Recovery in some cases would be unable to run algorithms in the correct order due to Operating System time precision, this is has been fixed by adding the execCount to the comment part of the scripts being built for recovery.
```
Comment (gvardany) [inline docs/source/release/v4.0.0/mantidplot.rst]: Do you mind removing the extra `is`?

Comment (Pasarus) [reply docs/source/release/v4.0.0/mantidplot.rst]: Done :+1:

**Review by martyngigg — APPROVED** [review]

**Pull request merged**